.pipelines/singletenancy/azure-cni-overlay/azure-cni-overlay-e2e-step-template.yaml

```diff
@@ -84,6 +84,51 @@ steps:
     displayName: "Get logs"
     condition: always()
 
+  - task: AzureCLI@1
+    inputs:
+      azureSubscription: $(AZURE_TEST_AGENT_SERVICE_CONNECTION)
+      scriptLocation: "inlineScript"
+      scriptType: "bash"
+      addSpnToEnvironment: true
+      inlineScript: |    
+        set -e
+        clusterName=${{ parameters.clusterName }}
+        echo "Restarting nodes"
+        for val in $(az vmss list -g MC_${clusterName}_${clusterName}_$(REGION_AKS_CLUSTER_TEST) --query "[].name" -o tsv); do
+          make -C ./hack/aks restart-vmss AZCLI=az CLUSTER=${clusterName} REGION=$(REGION_AKS_CLUSTER_TEST) VMSS_NAME=${val}
+        done
+    displayName: "Restart Nodes"
+
+  - task: AzureCLI@1
+    inputs:
+      azureSubscription: $(AZURE_TEST_AGENT_SERVICE_CONNECTION)
+      scriptLocation: "inlineScript"
+      scriptType: "bash"
+      addSpnToEnvironment: true
+      inlineScript: |
+        cd test/integration/load
+        clusterName=${{ parameters.clusterName }}
+        make -C ./hack/aks set-kubeconf AZCLI=az CLUSTER=${clusterName}
+        make -C ./hack/aks azcfg AZCLI=az REGION=$(REGION_AKS_CLUSTER_TEST)
+        kubectl get pods -owide -A
+        echo "Validating Node Restart"
+        sudo go test -timeout 30m -tags load -cni cniv2 -run ^TestValidateState$ -tags=load -restart-case=true
+    displayName: "Validate Node Restart"
+    retryCountOnTaskFailure: 3
+
+  - script: |
+      echo "validate pod IP assignment before CNS restart"
+      kubectl get pod -owide -A
+      kubectl rollout restart ds azure-cns -n kube-system
+      kubectl rollout status ds azure-cns -n kube-system
+      kubectl get pod -owide -A
+      echo "validate pods after CNS restart"
+      cd test/integration/load
+      sudo go test -timeout 30m -tags load -cni cniv2 -run ^TestValidateState$ -tags=load
```

Comment:
how does this test work? how do you validate that the state previous to restart is valid? this only looks at CNS state after restart, not before

Reply:
Same here

Reply:
> https://github.com/Azure/azure-container-networking/blob/master/.pipelines/singletenancy/azure-cni-overlay/azure-cni-overlay-e2e-step-template.yaml#L68

This line does not validate the cns state.

Reply:
added test validation before cns restarts